signer/src/transaction_coordinator.rs

```diff
@@ -2548,6 +2548,30 @@ pub async fn should_coordinate_dkg(
     let storage = context.get_storage();
     let config = context.config();
 
+    // Trigger dkg if signatures_required has changed
+    if let Some(last_key_rotation_signatures_required) = context
+        .get_storage()
+        .get_last_key_rotation(&bitcoin_chain_tip.block_hash)
+        .await?
+        .map(|last_key_rotation| last_key_rotation.signatures_required)
```

Comment:
We probably do not want to make this database read because it takes too long. Instead we want to add a new `signatures_required` field to the context's state object and just read it. We should set this new field in the `BlockObserver::set_signer_set_and_aggregate_key` function.

Reply:
@djordon two following commits should address it